.github/workflows/feature_deploy.yml

```diff
@@ -37,559 +37,151 @@ jobs:
-          az webapp deployment slot delete --name tfs-${{ env.environment }}-${{ env.repository }} --resource-group ${{ env.resource_group }} --slot ${{ github.sha }}
-
-    - name: 7. Restart
-      uses: azure/cli@v1.0.0
-      with:
-        inlineScript: |
-          az webapp restart --name tfs-${{ env.environment }}-${{ env.repository }} --resource-group ${{ env.resource_group }}
-
-  #################### 8. Numgen ####################
-  deploy-numgen:
-    needs: deployment-environment
-    name: Number generator
-    runs-on: [self-hosted, linux, deployment]
-    environment:
-      dev
-    env:
-      repository: azure-function-number-generator
-      path: azure-functions/number-generator-function
-      name: function-number-generator
-      
-    steps:
-    - name: 1. Latest commit hash
-      uses: actions/checkout@v3
-
-    - name: 2. Build and push
-      uses: docker/build-push-action@v1
-      with:
-        registry: ${{ env.acr }}
-        username: ${{ env.acr_username }}
-        password: ${{ env.acr_password }}
-        repository: ${{ env.repository }}
-        path: ${{ env.path }}
-        tags: latest
-        build_args: GITHUB_SHA=${{ github.sha }}
-        add_git_labels: true
-
-    - name: 3. ACR login
-      uses: azure/docker-login@v1
-      with:
-        login-server: ${{ env.acr }}
-        username: ${{ env.acr_username }}
-        password: ${{ env.acr_password }}
-
-    - name: 4. Tag images
-      run: |
-        from=${{ env.from }}
-        to=${{ env.environment }}
-        docker pull ${{ env.acr }}/${{ env.repository }}:$from
-        docker tag ${{ env.acr }}/${{ env.repository }}:$from ${{ env.acr }}/${{ env.repository }}:$to
-        docker push ${{ env.acr }}/${{ env.repository }}:$to
-
-    - name: 5. Azure login
-      uses: azure/login@v1
-      with:
-        creds: ${{ env.credentials }}
-
-    - name: 6. Slot
-      uses: azure/cli@v1.0.0
-      with:
-        inlineScript: |
-          az config get defaults
-          az functionapp deployment slot create --name tfs-${{ env.environment }}-${{ env.name }} --resource-group ${{ env.resource_group }} --slot ${{ github.sha }} --configuration-source tfs-${{ env.environment }}-${{ env.name }}
-          az functionapp deployment slot swap --name tfs-${{ env.environment }}-${{ env.name }} --resource-group ${{ env.resource_group }} --slot ${{ github.sha }} --action swap
-          az functionapp deployment slot delete --name tfs-${{ env.environment }}-${{ env.name }} --resource-group ${{ env.resource_group }} --slot ${{ github.sha }}
-
-    - name: 7. Restart
-      uses: azure/cli@v1.0.0
-      with:
-        inlineScript: |
-          az functionapp restart --name tfs-${{ env.environment }}-${{ env.name }} --resource-group ${{ env.resource_group }}
-
-  #################### 9. ACBS ####################
-  deploy-acbs:
-    needs: deployment-environment
-    name: ACBS
+      - name: 1. Latest commit hash
```

Comment:
The only difference is that I need to use `matrix.<blah>` for the matrix parameters.